### PR TITLE
Revamp rustler_mix documentation

### PR DESCRIPTION
**Pull request by evnu**
I restructured the documentation for rustler_mix in an attempt to make it more accessible on hex.pm. For this, I merged the basic guide into the `README.md`, made the `README.md` as the main file for `ex_doc`, replaced `def` with `defp` for internal functions, and removed internal modules from the output. Let me know what you think!

### CODE REVIEW

File: rustler_mix/README.md [hunk 1]
```diff
@@ -1,20 +1,99 @@
 # Rustler
 
-**TODO: Add description**
+This is the Mix package for [rustler](https://github.com/rusterlium/rustler), a library to write Erlang NIFs in
```
Comment (scrogson) [inline rustler_mix/README.md]: Sorry...I only meant in the URL 😀

File: rustler_mix/README.md [hunk 2]
```diff
@@ -1,20 +1,99 @@
...
+safe Rust code. Here, we provide the basic functionality to use rustler from Elixir:
```
Comment (scrogson) [inline rustler_mix/README.md]: s/rustler/Rustler

File: rustler_mix/README.md (hunk 1 above)
Comment (evnu) [reply rustler_mix/README.md]: :D

Comment (evnu) [reply rustler_mix/README.md]: 5bba2ea

**Pull request merged**